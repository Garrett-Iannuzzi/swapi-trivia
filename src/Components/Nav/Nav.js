import React from 'react';
import { Link } from 'react-router-dom';
import './Nav.scss';
import images from '../.././images/images.js'

const Nav = ({ userInfo, handleLoginError, resetUserInfo }) => {
  return (
    <nav>
      <div className='user-info'>
        <div className='user-inner'>
          <p className='i-p'>i</p>
          <p className='user-p'>{userInfo.name}</p>
          <p className='user-p'>{userInfo.quote}</p>
          <p className='user-p'>{userInfo.skillLevel}</p>
        </div>
      </div>
      <img src={images.starWarsLogo} alt="star wars logo" className="nav-logo" />
      <button className='btn-favorites'>Favorites</button>
      <Link to='/' className="link-logout">
        <button className='btn-logout' onClick={() => {resetUserInfo(); handleLoginError(false)}}>Logout</button>
      </Link>
    </nav>
  )
}

export default Nav;
