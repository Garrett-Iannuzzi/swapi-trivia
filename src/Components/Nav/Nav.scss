nav {
  border-bottom: solid 2px black;
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 99%;
}

.nav-logo {
  margin: 15px 0;
  height: 100px;
  width: 175px;
  opacity: .85;
}

.favorites-link {
  width: 100%;
  text-decoration: none;
}

.nav-favorites-img {
  height: 30px;
  width: 30px;
  margin: 0 10px;
}

.btn-favorites {
  opacity: .8;
  width: 100%;
  padding: 5px;
  border-radius: 10px;
  background-color: black;
  color: white;
  font-size: 1.5em;
  margin-bottom: 5px;
  outline: 0;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
}

.btn-favorites:hover {
  background-color: white;
  color: black;
}

.btn-logout {
  opacity: .8;
  width: 100%;
  padding: 5px;
  border-radius: 10px;
  background-color: black;
  color: white;
  font-size: 1.5em;
  outline: 0;
  cursor: pointer;
}

.btn-logout:hover {
  background-color: white;
  color: black;
}

.user-info {
  width: 450px;
  margin-left: 5px;
}

.user-inner {
  align-items: center;
  justify-content: space-between;
  display: flex;
  overflow: hidden;
  width: 90px;
  height: 100px;
  border-right: 2px solid black;
  transition: width 3s;
}

.nav-buttons {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  width: 15%;
}

.user-inner:hover {
  width: 500px;
}

.user-p {
  margin: 10px;
}

.link-logout {
  width: 100%;
}

span {
  font-weight: bold;
  text-decoration: underline;
}

.i-p {
  min-width: 70px;
  max-width: 70px;
  height: 70px;
  margin: 15px 10px;
  cursor: pointer;
  transition: transform 20s linear;
}

.i-p:hover {
  transition: transform 50s linear;
  transform: rotate(3600deg);
}
