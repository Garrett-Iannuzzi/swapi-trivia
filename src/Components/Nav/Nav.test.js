import React from 'react';
import { shallow } from 'enzyme';
import Nav from '../Nav/Nav.js';

describe('Nav', () => {

  let wrapper, mockUserInfo, mockHandleLoginError, mockResetUserInfo, mockFavorites

  beforeEach(() => {
    mockUserInfo = {
      name: 'Patrick',
      quote: 'Sick wiggy',
      skillLevel: 'Jedi Master'
    };
    mockHandleLoginError = jest.fn();
    mockResetUserInfo = jest.fn();
    favorites = {
        character: "C-3PO",
        creature: "artificial",
        name: "Tatooine",
        openingCrawl: "the guardians of",
        population: "200000",
        relatedFilms:[
          {relatedFilms: "The Empire Strikes Back"},
          {relatedFilms: "Attack of the Clones"}],
          species: "Droid"
      }
    wrapper = shallow(
      <Nav 
        userInfo={mockUserInfo}
        handleLoginError={mockHandleLoginError}
        resetUserInfo={mockResetUserInfo}
        favorites={mockFavorites}
      />
    )
  })




})