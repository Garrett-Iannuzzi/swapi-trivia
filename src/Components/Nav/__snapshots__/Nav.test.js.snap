// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Should match the snapshot with data passed in 1`] = `
<nav>
  <div
    className="user-info"
  >
    <div
      className="user-inner"
    >
      <img
        className="i-p"
        src="falcon.png"
      />
      <p
        className="user-p"
      >
        <span>
          Name: 
        </span>
        Patrick
      </p>
      <p
        className="user-p"
      >
        <span>
          Quote: 
        </span>
        Sick wiggy
      </p>
      <p
        className="user-p"
      >
        <span>
          Skill: 
        </span>
        Jedi Master
      </p>
    </div>
  </div>
  <img
    alt="star wars logo"
    className="nav-logo"
    src="starwarslogo.png"
  />
  <div
    className="nav-buttons"
  >
    <Link
      className="favorites-link"
      to="/favorites"
    >
      <button
        className="btn-favorites"
      >
        <img
          alt="heart icon"
          className="nav-favorites-img"
          src="lover.png"
        />
         Favorites: 
      </button>
    </Link>
    <Link
      className="link-logout"
      to="/"
    >
      <button
        className="btn-logout"
        onClick={[Function]}
      >
        Logout
      </button>
    </Link>
  </div>
</nav>
`;
